lib/Sema/CodeSynthesis.cpp

```diff
@@ -340,10 +340,10 @@ static FuncDecl *createMaterializeForSetPrototype(AbstractStorageDecl *storage,
   // Open-code the setMutating() calculation since we might run before
   // the setter has been type checked.
   Type contextTy = DC->getDeclaredInterfaceType();
-  materializeForSet->setMutating(
-      contextTy && !contextTy->hasReferenceSemantics() &&
+  if (contextTy && !contextTy->hasReferenceSemantics() &&
       !setter->getAttrs().hasAttribute<NonMutatingAttr>() &&
-      !storage->isSetterNonMutating());
+      !storage->isSetterNonMutating())
+    materializeForSet->setMutationModifier(MutationModifier::Mutating);
```

Comment:
We should figure out what to do if the getter or setter is `nonmutating` or `__consuming`.

Right now this logic breaks if the getter is mutating but the setter is non-mutating too (but that's not a useful combination I guess)